### DISCUSSION TITLE
How to custom filters items on button

### DISCUSSION THREAD

**Discussion by Korben26** [serenity-is/Serenity#6690] [Help]
Hello,
I've got a master and detail grids. When I select a master item, details is refreshed depending on the detail field "Num Groupe". It's work well.

But I would like to have items with "Num Groupe" NULL by clicking on button "voir les points de ventes non affectés".
![Sans titre](https://user-images.githubusercontent.com/124872881/218972942-c85a7d9b-68c5-49cb-b728-4968da0f2aea.png)

        protected getButtons() {
            return [{
                title: 'Voir les points de ventes non affectés',
                cssClass: 'expand-all-button',
                //onClick: () => this.view.setFilter ()
                //    [{
                //        getter: 'CategoryName'
                //    }])
            },
            {
                title: 'Sans les points de ventes non affectés',
                cssClass: 'collapse-all-button',
                //onClick: () => this.view.setGrouping([])
            }];
        }

How can I do that ?
Thanks for your answers

**Comment by Korben26** [comment]
Please I need some help !!

**Comment by wezmag** [comment]
Check out this.
https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.BasicSamples/Modules/Grids/GridFilteredByCriteria/GridFilteredByCriteria.ts

For `is null` criteria, use the following code.
```
["is null", ['YourFieldName']]
```
